pkg/query/endpointset.go

```diff
@@ -600,22 +550,6 @@ func (e *EndpointSet) GetEndpointStatus() []EndpointStatus {
 	return statuses
 }
 
-func (e *EndpointSet) cleanUpEndpointStatuses(endpoints map[string]*endpointRef) {
```

Comment:
Any idea why this was here? From what I understand, strict endpoints are always used, and non-strict endpoints are removed when info calls fail.

Reply:
Ok figured it out. We actually want to keep unhealthy endpoints around for a while for the purpose of showing them as `DOWN` in the UI. 

Update the PR and added some more tests for this use case.

Reply:
👍🏽